spec/ruby/core/method/shared/eql.rb

```diff
@@ -54,6 +54,16 @@
     @m_foo.send(@method, m2).should be_true
   end
 
+  it "returns false if comparing a method defined via define_method and def" do
+    MethodSpecs::Methods.send :define_method, :defined_foo, lambda {}
```

Comment:
I think we should avoid modifying the fixtures from within the spec bodies.
It looks like `MethodSpecs::Methods` already has [some methods defined with `define_method`](https://github.com/rubinius/rubinius/blob/621953b6e6ea1e948de8e70ac3422a98f3a3c82f/spec/ruby/core/method/fixtures/classes.rb#L80-L85).  You could probably use one of those, or move your defined method to that file.